Fixes for gdal 0.18.x and change shuffle.

gregersn opened the pull request:
I have probably something done stuff that is very non-rustic, as I am not a rust coder.
But this seems to fix both the update of gdal and the colors shuffle issue.

Close #53

nms-scribe commented:
I appreciate all the work you did. 

It's really difficult for me to see what you changed in with all of the formatting changes you made. Some of that reformatting probably needed to be done, but I'd prefer that were in a different commit so I can check it over. Any chance you could go back and put those things back the way they were?

For the shuffle (lines 401 and 402 in your version of colors.rs), I'd prefer you use the `rng` variable I already have in scope, no need to create a new random number generator. The idea behind reusing it is to allow the same seed to produce the same results -- at least when the binary and the machine hasn't changed. I know the code doesn't always work that way, but it's the goal. Also, you deleted the comment above it explaining what's going on.

For the field name access, what I see looks right. Eventually, I should go to using indexes directly somehow, but that would probably require a lot of reworking to my typed feature abstractions.

gregersn commented:
All sensible comments.
I have split formatting into a separate commit, and also updated commit message, etc. 
I hope the formatting is okay, but from what I understood from other files I had open, that did not change, the auto formatting probably has sensisble defaults?

gregersn committed:
Fixes for gdal 0.18.x and change shuffle.

gdal 0.18.x has removed most functions for dealing with fields directly
by name, so we have to get their index where fields are used.

The color randomization misused Vec::sort to shuffle, which newer rust
does not like, as it didn't implement a "total order". Switch to using
rand::seq::SliceRandom::shuffle instead.

Also scope mutable borrow of tile_layer.
